app/models/spree/adjustment.rb

```diff
@@ -70,6 +70,7 @@ class Adjustment < ApplicationRecord
     scope :credit, -> { where('amount < 0') }
     scope :return_authorization, -> { where(originator_type: "Spree::ReturnAuthorization") }
     scope :voucher, -> { where(originator_type: "Voucher") }
+    scope :voucher_tax, -> { where(originator_type: "Voucher").where("label LIKE 'Tax%'") }
```

Comment:
In [update_tax_adjustment_for](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/services/voucher_adjustments_service.rb#L53) we're currently defining the conditions for finding a tax adjustment on an order, and this rule is much less specific. In particular it makes me uncomfortable that we have to partial match a label (what if someone manually creates an adjustment labeled "Tax"? Should we make it a reserved word?). 
Also uncomfortable using a `first` below.

I think perhaps the VoucherAdjustmentsService should have a method to find the order's voucher tax adjustment, which allows us to be more explicit about how many records may be found (zero or one for now, but maybe multiple in the future if we have multiple vouchers per order?)

What do you think, does that make sense?

Reply:
🤔 Hmm.. or would this be better modelled with standard ActiveModel features? I forget why we don't have a model for VoucherAdjustment..

Reply:
Yeah I think the the modelling of Voucher Tax adjustment isn't great at the moment and should probably be fixed ( Maikel raised similar concerns ) . I'll look into it a bit more, maybe we could use the `voucher_adjustment` as an `originator_type` for the adjustment that stores the tax part of the voucher when tax are excluded from price.

To answer your question, the label Tax is problematic, one could create a voucher called "Tax123" and it would break the current implementation. Currently, it is only possible to add one voucher to an order, at least via the UI, hence the use of `first` here. It's not restricted by the code. so technically you should be able to have more than one voucher per order.

Reply:
@maikel @dacook What about using `AdjustmentMetadata` ?
We could use `fee_name: Tax, fee_type: "Voucher"` in the metadata associated with the `voucher_adjustment` describing the tax part. Then you can do something like : 
```ruby
order.voucher_adjustments.joins(:metadata).where(metadata: { fee_name: "Tax", fee_type: "Voucher" })
```
That way we still keep the voucher data as an order adjustment but we get rid of the ugly label match : `"label LIKE 'Tax%'"`

Reply:
I'm not familiar with AdjustmentMetadata yet so can't comment yet.. Maikel if you can comment please do, otherwise let me know and I'll look into it.

Reply:
As far as I can see it's only used in reports : https://github.com/openfoodfoundation/openfoodnetwork/blob/693d5bad4aa610e13b3b366172ebd727710207a6/lib/reporting/reports/enterprise_fee_summary/scope.rb#L69C1-L76

Reply:
Yes, it's another join but if we need to do that then we should. It's better then relying on the label.

Reply:
Fixed here : https://github.com/openfoodfoundation/openfoodnetwork/pull/11543/commits/03ce39d5c536fc0db41951dd1d185b2c423c883e  no more hacky `LIKE` matching